packages/commerce-sdk-react/src/auth/storage.ts

```diff
@@ -44,16 +45,28 @@ export class CookieStorage extends BaseStorage {
         super(options)
 
         if (typeof document === 'undefined') {
-            throw new Error('CookieStorage is not avaliable on the current environment.')
+            throw new Error('CookieStorage is not available on the current environment.')
         }
     }
     set(key: string, value: string, options?: Cookies.CookieAttributes) {
         const suffixedKey = this.getSuffixedKey(key)
+        const parentUrl = document.location?.ancestorOrigins?.[0] || document.referrer
```

Comment:
Minor: how about extracting this sameSite logic into a separate function? It feels distracting to have it here right now, since it talks about parent url, iframe, etc.

Once extracted, we can have a cleaner `set` method like this:

```ts
    set(key: string, value: string, options?: Cookies.CookieAttributes) {
        const suffixedKey = this.getSuffixedKey(key)
        Cookies.set(suffixedKey, value, {
            // Deployed sites will always be HTTPS, but the local dev server is served over HTTP.
            // Ideally, this would be `secure: true`, because Chrome and Firefox both treat
            // localhost as a Secure context. But Safari doesn't, so here we are.
            secure: !onClient() || window.location.protocol === 'https:',
            sameSite: getCookieSameSiteAttribute(),
            ...options
        })
    }
```

Reply:
Good suggestion. I've moved it to util file for cookieSameSite attribute